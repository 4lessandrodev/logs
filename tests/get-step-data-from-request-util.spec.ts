import { getStepDataFromRequest } from "../lib/utils/get-step-data-from-request.util"

describe('get-step-data-from-request.util', () => {
    it('should get step data with success', () => {
        const error: Error = { message: 'Error Message', name: 'Error', stack: 'Error Stack' };
        const request: any = { method: "POST", body: {} };
        const result = getStepDataFromRequest(error, request);
        expect(result).toEqual({
            "body": {},
            "data": "{}",
            "message": "Error Message",
            "method": "POST",
            "stack": "Error Stack",
            "statusCode": 500,
            "tags": [],
            "uid": undefined
        });
        expect(result).toMatchSnapshot();
    });

    it('should get step data with success', () => {
        const error: Error = { message: 'Error Message', name: 'Error', stack: 'Error Stack' };
        const request: any = { method: "POST", body: { email: "some@mail.com", id: "1" } };
        const result = getStepDataFromRequest(error, request);
        expect(result).toEqual({
            uid: "1",
            message: 'Error Message',
            stack: 'Error Stack',
            statusCode: 500,
            method: 'POST',
            tags: ["email", "id"],
            data: `{
  \"email\": \"some@mail.com\",
  \"id\": \"1\"
}`,
            body: { email: "some@mail.com", id: "1" }
        });
        expect(result).toMatchSnapshot();
    });

    it('should get all default', () => {
        const error = {};
        const request: any = {};
        const result = getStepDataFromRequest(error as any, request);
        expect(result).toEqual({
            message: 'Internal Server Error',
            stack: '',
            statusCode: 500,
            method: 'NONE',
            tags: [],
            data: "{}",
            body: {}
        });
        expect(result).toMatchSnapshot();
    });

    it('should get empty tags if body is array', () => {
        const error = {};
        const request: any = { body: [1,2,3] };
        const result = getStepDataFromRequest(error as any, request);
        expect(result).toEqual({
            message: 'Internal Server Error',
            stack: '',
            statusCode: 500,
            method: 'NONE',
            tags: [],
            data: "{}",
            body: [
                1,
                2,
                3
            ],
            uid: undefined,
        });
        expect(result).toMatchSnapshot();
    });
})