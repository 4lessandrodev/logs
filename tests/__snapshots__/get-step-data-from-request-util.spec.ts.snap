// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-step-data-from-request.util should get all default 1`] = `
{
  "body": {},
  "data": "{}",
  "message": "Internal Server Error",
  "method": "NONE",
  "stack": "",
  "statusCode": 500,
  "tags": [],
  "uid": undefined,
}
`;

exports[`get-step-data-from-request.util should get empty tags if body is array 1`] = `
{
  "body": [
    1,
    2,
    3,
  ],
  "data": "{}",
  "message": "Internal Server Error",
  "method": "NONE",
  "stack": "",
  "statusCode": 500,
  "tags": [],
  "uid": undefined,
}
`;

exports[`get-step-data-from-request.util should get step data with success 1`] = `
{
  "body": {},
  "data": "{}",
  "message": "Error Message",
  "method": "POST",
  "stack": "Error Stack",
  "statusCode": 500,
  "tags": [],
  "uid": undefined,
}
`;

exports[`get-step-data-from-request.util should get step data with success 2`] = `
{
  "body": {
    "email": "some@mail.com",
    "id": "1",
  },
  "data": "{
  "email": "some@mail.com",
  "id": "1"
}",
  "message": "Error Message",
  "method": "POST",
  "stack": "Error Stack",
  "statusCode": 500,
  "tags": [
    "email",
    "id",
  ],
  "uid": "1",
}
`;
